components/org.wso2.carbon.crypto.provider.hsm/src/main/java/org/wso2/carbon/crypto/provider/hsm/cryptoprovider/util/MechanismResolver.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util;
+
+import iaik.pkcs.pkcs11.Mechanism;
+import iaik.pkcs.pkcs11.parameters.GcmParameters;
+import iaik.pkcs.pkcs11.parameters.InitializationVectorParameters;
+import iaik.pkcs.pkcs11.parameters.RSAPkcsOaepParameters;
+import iaik.pkcs.pkcs11.parameters.RSAPkcsPssParameters;
+import iaik.pkcs.pkcs11.wrapper.PKCS11Constants;
+import org.wso2.carbon.crypto.api.CryptoException;
+
+import java.security.SecureRandom;
+import java.util.HashMap;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+
+import static org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util.CryptoConstants.DECRYPT_MODE;
+import static org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util.CryptoConstants.ENCRYPT_MODE;
+
+/**
+ * This class is used to resolve JCE standard mechanism names to PKCS #11 mechanisms.
+ */
+public class MechanismResolver {
+
+    private static MechanismResolver mechanismResolver;
+    private static SecureRandom random = new SecureRandom();
+    private static HashMap<String, Long> mechanisms = new HashMap<String, Long>() {{
```

Comment:
Is there any special reason to have all these in a map ? Is this a conversion ?

Reply:
This map converts JCE standard algorithms to PKCS #11 algorithms.